tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -375,7 +375,8 @@ std::vector<std::unique_ptr<TChain>> MakeFriends(const ROOT::TreeUtils::RFriendI
          // Otherwise, the new friend chain needs to be built using the nomenclature
          // "filename?#treename" as argument to `TChain::Add`
          for (std::size_t j = 0u; j < thisFriendFileNames.size(); ++j) {
-            frChain->Add((thisFriendFileNames[j] + "?#" + thisFriendChainSubNames[j]).c_str(), thisFriendEntries[j]);
+            // Use the old syntax '/' to play around https://github.com/root-project/root/issues/11483
+            frChain->Add((thisFriendFileNames[j] + "/" + thisFriendChainSubNames[j]).c_str(), thisFriendEntries[j]);
```

Comment:
I'm afraid this would introduce a regression: as far as I understand we were using `?#` here because `chain->Add("filename/treename")` is ambiguous when filename does not end in `.root`: in that case `TChain` interprets its argument as the full path to the file, with no treename specified. See also https://github.com/root-project/root/issues/8739 .

If `thisFriendFileNames` can contain globs (incurring in #11483 ) then one possible workaround is to first expand the globs to a list of concrete file names and then call `chain->Add` with the `?#` syntax.

But anyway I don't understand how the first 2 commits in this PR would trigger #11483 ?

Reply:
They trigger #11483 in `dataframe_datasetspec.cxx` tests, when the `RLoopManager(RDatasetSpec &&spec)` constructor is used and the friends in the spec have file names with globs that incur in that issue.

> If thisFriendFileNames can contain globs (incurring in https://github.com/root-project/root/issues/11483 ) then one possible workaround is to first expand the globs to a list of concrete file names and then call chain->Add with the ?# syntax.

I don't know how we can expand the globs without creating temporary TChains everytime, need to think about it.

Reply:
Yes, the simplest way is to create a temporary TChain. I don't think it adds much overhead to the existing machinery, but then again if there is a better way.. :)

Reply:
The better way is at https://github.com/root-project/root/pull/12284 . Once that's in, this PR can probably be merged